src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -123,12 +123,12 @@ private async ValueTask ProcessWorkspaceChangeAsync(CancellationToken cancellati
                 return;
 
             var trackingService = _workspace.Services.GetRequiredService<IDocumentTrackingService>();
-            var priorityDocument = trackingService.TryGetActiveDocument();
+            var priorityDocumentId = trackingService.GetActiveDocument(solution)?.Id;
```

Comment:
this ensures that hte priority doc id actually belongs to the solution snapshot we're looking at.